oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/register/service/IServiceInventoryRegister.java

```diff
@@ -35,5 +35,5 @@
 
     void heartbeat(int serviceId, long heartBeatTime);
 
-    void updateMapping(int serviceId, int mappingServiceId);
+    void updateMapping(int serviceId, int mappingServiceId, boolean force);
```

Comment:
I would recommend you to provide a `#resetMapping` method. Because in that case, there is no need to provide `#mappingServiceId`. This method only should be called when you want to mappingServiceId back to 0. In other cases, this update works w/o `force==true`.

This change seems to misguide the API user, if `force` usually means, update even the value is there(not null/0), but here, your meaning is, mappingServiceId could be set as 0 when the mappingServiceId != 0.